packages/@aws-cdk/aws-apigatewayv2/lib/http/stage.ts

```diff
@@ -176,4 +176,18 @@ export class HttpStage extends HttpStageBase {
     const urlPath = this.stageName === DEFAULT_STAGE_NAME ? '' : this.stageName;
     return `https://${this.api.apiId}.execute-api.${s.region}.${s.urlSuffix}/${urlPath}`;
   }
+
+  /**
+   * The custom domain URL to this stage
+   */
+  public get domainUrl(): string {
+    if (!this.apiMapping) {
+      throw new Error('domainUrl is not available when domain mapping has not been added');
+    }
+    if (!this.apiMapping.domainName.name) {
+      throw new Error('Unable to build domainUrl due to invalid domainName');
+    }
```

Comment:
Shouldn't need this anymore now that `domainName` is not optional, right?

If the check is for domainName being empty string (indicated by your test case), 
(a) I wonder if empty string domain names are valid strings. Perhaps disallow it in the CDK?
(b) update the message to be more clear, i.e., what is invalid about it.

Reply:
It doesn't appear that the aws cli command for creating an API Mapping allows for an empty string after some testing, so I will throw an error in the ApiMapping constructor if an empty string was passed for the name field

Reply:
Yes, that would be preferrable.

Reply:
@nija-at I had updated my comment perhaps after you read it. Which class constructor should I put this error in? ApiMapping or DomainName?

DomainName would catch the error earlier, but I want to check with you before I proceed.

Reply:
DomainName sounds like the right spot.